src/tools/run-make-support/src/rustdoc.rs

```diff
@@ -0,0 +1,59 @@
+use std::path::Path;
+use std::process::{Command, Output};
+
+use crate::{handle_failed_output, setup_common_build_cmd};
+
+pub fn rustdoc() -> RustdocInvocationBuilder {
+    RustdocInvocationBuilder::new()
+}
+
+#[derive(Debug)]
+pub struct RustdocInvocationBuilder {
+    cmd: Command,
+}
+
+impl RustdocInvocationBuilder {
+    fn new() -> Self {
+        let cmd = setup_common_build_cmd();
```

Comment:
Actually, this might not be right, `setup_common_build_cmd()` is unforuntately named but it only sets up common build command for `rustc` and `aux_build` invocations, it doesn't set it up for rustdoc. You probably want to change this to `setup_common_rustc_build_cmd()` or whatever and add rustdoc setups here instead.